src/main/java/devybot/TaskList.java

```diff
@@ -0,0 +1,164 @@
+package devybot;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.util.ArrayList;
+
+import devybot.exceptions.EmptyDescriptionException;
+import devybot.exceptions.TaskIndexOutOfBoundsException;
+import devybot.tasks.DeadlineTask;
+import devybot.tasks.EventTask;
+import devybot.tasks.Task;
+import devybot.tasks.TodoTask;
+
+public class TaskList {
+    private ArrayList<Task> taskList;
+
+    public TaskList() {
+        this.taskList = new ArrayList<>();
+    }
+
+    public TaskList(ArrayList<Task> tasks) {
+        this.taskList = tasks;
+    }
+
+    public Task getTask(int index) {
+        return this.taskList.get(index);
+    }
+
+    public void deleteTask(int index) {
+        try {
+            if (index >= this.taskList.size() || index < 0) {
+                throw new TaskIndexOutOfBoundsException(index);
+            }
+            Task currentTask = this.taskList.get(index);
+            taskList.remove(index);
+            String outpString = "Noted. I've removed this task:\n  " + currentTask;
+            Ui.showMessage(outpString);
+        } catch (TaskIndexOutOfBoundsException e) {
+            Ui.showMessage(e.getMessage());
+        }
+    }
+
+    public int size() {
+        return this.taskList.size();
+    }
+
+    public void listTasks() {
+        if (taskList.size() == 0) {
+            Ui.showMessage("Currently no tasks available.");
+            return;
+        }
+        String outpString = "Here are the tasks in your list:\n";
+        for (int i = 0; i < taskList.size(); i++) {
+            outpString += (i + 1) + ". " + taskList.get(i).toString() + "\n";
+        }
+        Ui.showMessage(outpString);
+    }
+
+    public void addTodoTask(String userInput) throws EmptyDescriptionException {
+        String description = userInput.substring(4).trim();
+        if (description.isEmpty()) {
+            throw new EmptyDescriptionException("todo");
+        }
+
+        Task newTask = new TodoTask(description);
+        taskList.add(newTask);
+
+        String outpString = "Got it. I've added this task:\n  " + newTask + "\nNow you have " + taskList.size()
+                + " tasks in the list.";
+        Ui.showMessage(outpString);
+    }
+
+    public void addDeadlineTask(String userInput) throws EmptyDescriptionException {
```

Comment:
Perhaps should consider abstracting the logic for parsing and adding a task, right now you have duplicate code in both the `addDeadlineTask()` and `addEventTask()`, as well as in your `Storage` class